modules/router/src/services/fees.ts

```diff
@@ -334,199 +337,210 @@ export const calculateFeeAmount = async (
-    return Result.fail(
-      new FeeError(FeeError.reasons.ChannelError, {
-        message: "Not in channel",
-        publicIdentifier: routerPublicIdentifier,
-        alice: toChannel.aliceIdentifier,
-        bob: toChannel.bobIdentifier,
-        channelAddress: toChannel.channelAddress,
-      }),
-    );
-  }
-  const routerBalance = getBalanceForAssetId(toChannel, toAssetId, participantToChannel);
-  // get the amount you would send
-  const isSwap = fromAssetId !== toAssetId || fromChannel.networkContext.chainId !== toChannel.networkContext.chainId;
-  const converted = isSwap
-    ? await getSwappedAmount(
-        amountToSend.toString(),
-        fromAssetId,
-        fromChannel.networkContext.chainId,
-        toAssetId,
-        toChannel.networkContext.chainId,
-      )
-    : Result.ok(amountToSend.toString());
-  if (converted.isError) {
-    return Result.fail(
-      new FeeError(FeeError.reasons.ConversionError, {
-        swapError: jsonifyError(converted.getError()!),
-      }),
-    );
-  }
-  if (BigNumber.from(routerBalance).gte(converted.getValue())) {
-    // channel has balance, no extra gas required to facilitate transfer
+  if (!toChannel) {
+    // if no channel exists, we need to account for the full channel deployment always
+    // NOTE: same assumption as above for alice
+    return Result.ok([fromChannelFee, GAS_ESTIMATES.createChannelAndDepositAlice]);
+  } else {
+    const participantToChannel = getParticipant(toChannel, routerPublicIdentifier);
+    if (!participantToChannel) {
+      return Result.fail(
+        new FeeError(FeeError.reasons.ChannelError, {
+          message: "Not in channel",
+          publicIdentifier: routerPublicIdentifier,
+          alice: toChannel.aliceIdentifier,
+          bob: toChannel.bobIdentifier,
+          channelAddress: toChannel.channelAddress,
+        }),
+      );
+    }
+    const routerBalance = getBalanceForAssetId(toChannel, toAssetId, participantToChannel);
+    // get the amount you would send
+    const isSwap = fromAssetId !== toAssetId || fromChainId !== toChainId;
+    const converted = isSwap
+      ? await getSwappedAmount(amountToSend.toString(), fromAssetId, fromChainId, toAssetId, toChainId)
+      : Result.ok(amountToSend.toString());
+    if (converted.isError) {
+      return Result.fail(
+        new FeeError(FeeError.reasons.ConversionError, {
+          swapError: jsonifyError(converted.getError()!),
+        }),
+      );
+    }
+    if (BigNumber.from(routerBalance).gte(converted.getValue())) {
+      // channel has balance, no extra gas required to facilitate transfer
+      logger.info(
+        { method, methodId, routerBalance: routerBalance.toString(), amountToSend: amountToSend.toString() },
+        "Channel is collateralized",
+      );
+      logger.debug(
+        {
+          method,
+          methodId,
+        },
+        "Method complete",
+      );
+      return Result.ok([fromChannelFee, Zero]);
```

Comment:
Reclamation?

Reply:
I didn't change this, is it missing from the original code?

Reply:
Could be